File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/direct/Java/Python/s378126458_commented.py
import sys

def main():
    input = sys.stdin.read
    data = input().split()
    index = 0
    N = int(data[index])
    index += 1
    Q = int(data[index])
    index += 1
    s = data[index]
    index += 1
    pre = [0] * (N + 1)
    
    for i in range(1, N):
        if s[i-1] == 'A' and s[i] == 'C':
            pre[i] = 1  # initially the prefix sum array is the k array
    
    # Building the psa
    for i in range(1, N):
        pre[i] += pre[i-1]
    
    # Querying
    for _ in range(Q):
        l = int(data[index])
        index += 1
        r = int(data[index])
        index += 1
        r -= 1
        print(pre[r] - pre[l-1])

main()  # End of Code

File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/direct/Java/C++/s378126458_commented.cpp
#include <iostream>
#include <vector>
#include <string>

int main() {
    int N, Q;
    std::cin >> N >> Q;
    std::string s;
    std::cin >> s;
    std::vector<int> pre(N + 1, 0);
    for (int i = 1; i < N; i++) {
        if (s[i - 1] == 'A' && s[i] == 'C') {
            pre[i] = 1; // initially the prefix sum array is the k array
        }
    }
    // Building the psa
    for (int i = 1; i < N; i++) {
        pre[i] += pre[i - 1];
    }
    // Querying
    while (Q-- > 0) {
        int l, r;
        std::cin >> l >> r;
        r--;
        std::cout << pre[r] - pre[l - 1] << std::endl;
    }
}
// End of Code
